server/controllers/category.js

```diff
@@ -1,40 +1,37 @@
-import Category from '../models/category.js';
-import slugify from 'slugify';
+import Category from "../models/category.js"
+import slugify from "slugify"
+import { validateCreate,validateUpdate } from '../validators/category/categoryValidation.js';
 
 export const create = async (req, res) => {
-  try {
-    const { name } = req.body;
-    if (!name.trim()) {
-      return res.json({ error: 'Name is required' });
-    }
-    const existingCategory = await Category.findOne({ name });
-    if (existingCategory) {
-      return res.json({ error: 'Already exist' });
+    try {
+        const { name } = req.body;
+        const error = await validateCreate({name});
+        if (error.length > 0) {
```

Comment:
```suggestion
        if (error.length) {
```

poderia testar sem a condicao do > 0, acredito que quando ele não tiver erro no array o length do array vai se comportar como 0 que eh igual a falso no boolean